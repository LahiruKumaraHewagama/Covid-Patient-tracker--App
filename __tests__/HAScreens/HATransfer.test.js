import React from "react";
import HospitalAdminTransfer from "../../screens/HospitalAdminScreens/HospitalAdminTransfer";
import { act, create } from "react-test-renderer";
import { render, fireEvent, waitFor, cleanup } from '@testing-library/react-native';

jest.useFakeTimers();
const navigation = { addListener: jest.fn() }
const route = { params: { id: "" } }
afterEach(cleanup);

let tree
test('snapshot HA Transfer page', async () => {
    act(() => { tree = create(<HospitalAdminTransfer navigation={navigation} route={route} />) })
    expect(tree).toMatchSnapshot();
})

test('Id empty input error massage', async () => {
    global.alert = jest.fn();
    const transferBtn = tree.root.findByProps({ testID: 'transfer' }).props;
    act(() => transferBtn.onPress());
    expect(global.alert).toHaveBeenCalledWith("Id can't be empty !");
});

test('Destination empty input error massage', async () => {
    global.alert = jest.fn();
    const id = tree.root.findByProps({ testID: 'id_test' }).props;
    act(() => id.onChangeText("12636353574430220"));
    const transferBtn = tree.root.findByProps({ testID: 'transfer' }).props;
    act(() => transferBtn.onPress());
    expect(global.alert).toHaveBeenCalledWith("Please select Destination BedID !");
});


test('Successfully Transfer', async () => {
    global.alert = jest.fn();
    const id = tree.root.findByProps({ testID: 'id_test' }).props;
    act(() => id.onChangeText("12636353574430220"));

    const dest = tree.root.findByProps({ testID: 'dest_test' }).props;
    act(() => dest.onChangeText("12"));

    const transferBtn = tree.root.findByProps({ testID: 'transfer' }).props;
    act(() => transferBtn.onPress());
    expect(global.alert.length).toEqual(0);
});
