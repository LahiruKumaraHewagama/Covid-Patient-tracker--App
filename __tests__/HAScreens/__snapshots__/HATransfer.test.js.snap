// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot HA Transfer page 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 3,
      "paddingHorizontal": 20,
      "paddingVertical": 20,
    }
  }
>
  <View
    style={
      Object {
        "justifyContent": "flex-end",
        "paddingBottom": 20,
        "paddingHorizontal": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#009387",
          "fontSize": 30,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      TRANSFER  PATIENT 
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#009387",
          "borderBottomWidth": 2,
        }
      }
    />
  </View>
  <RCTScrollView
    style={
      Object {
        "marginTop": 50,
        "paddingRight": 20,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#007c7a",
            "fontSize": 16,
          }
        }
      >
        Patient ID
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#007c7a",
            "borderRadius": 20,
            "borderWidth": 1,
            "flexDirection": "row",
            "marginTop": 5,
            "padding": 5,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="Enter Id"
          placeholderTextColor="#666666"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 15,
              "paddingLeft": 20,
            }
          }
          testID="id_test"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <Text
        style={
          Object {
            "color": "#007c7a",
            "fontSize": 16,
            "paddingTop": 15,
          }
        }
      >
        Destination Bed Id
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#007c7a",
            "borderRadius": 20,
            "borderWidth": 1,
            "flexDirection": "row",
            "marginTop": 5,
            "padding": 5,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          keyboardType="numeric"
          onChangeText={[Function]}
          placeholder="Enter Destination Bed Id"
          placeholderTextColor="#666666"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 15,
              "paddingLeft": 20,
            }
          }
          testID="dest_test"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#009387",
            "borderColor": "#20d1ce",
            "borderRadius": 10,
            "borderWidth": 2,
            "marginBottom": 20,
            "marginLeft": 40,
            "marginRight": 40,
            "marginTop": 60,
            "opacity": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 20,
            }
          }
        >
          Transfer
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
